docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1058 @@
+        .Run(args, config);
+}
+```
+
+**Note:** the config defines a [disassembler](https://adamsitnik.com/Disassembly-Diagnoser/), which exports a disassembly in GitHub markdown format (supported on both x64 and arm64, Windows and Linux). It is very often an invaluable tool when working with high-performance code where inspecting generated assembly code is required.
+
+#### Memory alignment
+
+BenchmarkDotNet does a lot of heavy lifting for the end users, but it can not protect us from the random memory alignment which can be different per each benchmark run and affect the stability of the benchmarks.
+
+We have three possibilities:
+
+* We can enforce the alignment ourselves and have very stable results.
+* We can ask the harness to try to randomize the memory and observe entire possible distribution with each run.
+* We can do nothing and wonder why the results vary from time to time.
+
+##### Enforcing memory alignment
+
+We can allocate aligned unmanaged memory by using the [NativeMemory.AlignedAlloc](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.nativememory.alignedalloc).
+
+```cs
+public unsafe class Benchmarks
+{
+    private void* _pointer;
+
+    [Params(6, 32, 1024)] // test various sizes
+    public uint Size;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        _pointer = NativeMemory.AlignedAlloc(byteCount: Size * sizeof(int), alignment: 32);
+        NativeMemory.Clear(_pointer, byteCount: Size * sizeof(int)); // ensure it's all zeros, so 1 is never found
+    }
+
+    [Benchmark]
+    public bool Contains()
+    {
+        ReadOnlySpan<int> buffer = new (_pointer, (int)Size);
+        return buffer.Contains(1);
+    }
+
+    [GlobalCleanup]
+    public void Cleanup() => NativeMemory.AlignedFree(_pointer);
+}
+```
+
+Sample results (please mind the AVX2, AVX and SSE4.2 information printed in the summary):
+
+```ini
+BenchmarkDotNet=v0.13.5, OS=Windows 11 (10.0.22621.1413/22H2/2022Update/SunValley2)
+AMD Ryzen Threadripper PRO 3945WX 12-Cores, 1 CPU, 24 logical and 12 physical cores
+.NET SDK=8.0.100-alpha.1.22558.1
+  [Host]    : .NET 7.0.4 (7.0.423.11508), X64 RyuJIT AVX2
+  Scalar    : .NET 7.0.4 (7.0.423.11508), X64 RyuJIT
+  Vector128 : .NET 7.0.4 (7.0.423.11508), X64 RyuJIT AVX
+  Vector256 : .NET 7.0.4 (7.0.423.11508), X64 RyuJIT AVX2
+```
+
+```
+|   Method |       Job | Size |       Mean |    StdDev | Ratio | Code Size |
+|--------- |---------- |----- |-----------:|----------:|------:|----------:|
+| Contains |    Scalar | 1024 | 143.844 ns | 0.6234 ns |  1.00 |     206 B |
+| Contains | Vector128 | 1024 | 104.544 ns | 1.2792 ns |  0.73 |     335 B |
+| Contains | Vector256 | 1024 |  55.769 ns | 0.6720 ns |  0.39 |     391 B |
+```
+
+The results should be very stable (flat distributions), but on the other hand we are measuring the performance of best case scenario (the input is large and it's entire content is searched for, as the value is never found).
```

Comment:
```suggestion
The results should be very stable (flat distributions), but on the other hand we are measuring the performance of the best case scenario (the input is large and its entire contents is searched through, as the value is never found).
```

Reply:
@Rob-Hague are you sure that `is` should still be used here?:

```diff
- its entire contents is searched through
+ its entire contents are searched through
```

?